# HG changeset patch
# User Rik <rik@octave.org>
# Date 1362170816 28800
#      Fri Mar 01 12:46:56 2013 -0800
# Node ID 8650eec57e9f05c718d6d2d04886ec8005ddfafc
# Parent  22bb3aa9f025c1e148ecb3d78bd29b32ec48cae7
Set LC_NUMERIC to "C" locale regardless of user value. (Matlab compatibility and bug #38372)

* libgui/src/octave-adapter/octave-main-thread.cc (run):
Set LC_NUMERIC to "C" locale.

diff --git a/libgui/src/octave-adapter/octave-main-thread.cc b/libgui/src/octave-adapter/octave-main-thread.cc
--- a/libgui/src/octave-adapter/octave-main-thread.cc
+++ b/libgui/src/octave-adapter/octave-main-thread.cc
@@ -35,16 +35,18 @@ along with Octave; see the file COPYING.
 octave_main_thread::octave_main_thread () : QThread ()
 {
 }
 
 void
 octave_main_thread::run ()
 {
   setlocale (LC_ALL, "en_US.UTF-8");
+  // Matlab uses "C" locale for LC_NUMERIC class regardless of local setting
+  setlocale (LC_NUMERIC, "C");
 
   emit ready ();
 
   octave_initialize_interpreter (octave_cmdline_argc, octave_cmdline_argv,
                                  octave_embedded);
 
   octave_execute_interpreter ();
 }
